14_contains/solution/contains-solution.spec.js

```diff
@@ -0,0 +1,45 @@
+const contains = require('./contains-solution');
+
+describe('contains', () => {
+  const nestedObject = {
+    data: {
+      banana: NaN,
+      info: {
+          magicNumber: 44,
+          stuff: {
+              thing: {
+                banana: NaN,
+                moreStuff: {
+                    something: 'foo',
+                    answer: [42]
+                }
+              }
+          }
+      }
+    }
+  }
+
+  test('true if property is number', () => {
+    expect(contains(nestedObject, 44)).toBe(true);
+  });
+
+  test('true if property is string', () => {
+    expect(contains(nestedObject, 'foo')).toBe(true);
+  });
+
+  test('does not do type conversions', () => {
+    expect(contains(nestedObject, '44')).toBe(false);
+  });
+
+  test('false if property is not in object', () => {
+    expect(contains(nestedObject, 'bar')).toBe(false);
+  });
+
+  test('true if property is duplicated', () => {
+    expect(contains(nestedObject, NaN)).toBe(true);
```

Comment:
`NaN` is better tested in its own test because NaN equality is a special case (`NaN === NaN` evaluates to `false` by definition) and would require specific `NaN` equality checks, or a method that has that build in (like `.includes`) e.g.
```js
test('true if NaN is a value within the object', () => {
  expect(contains(object, NaN)).toBe(true);
});
```

A `true if provided value is within the object, even if duplicated` test (wording following previous point about test descriptions) would better use a normal primitive, meaning you should change the test object to duplicate something different.